liblwgeom/lwgeom_transform.c

```diff
@@ -44,33 +44,67 @@ to_dec(POINT4D *pt)
 	pt->y *= 180.0/M_PI;
 }
 
+/***************************************************************************/
+
+#if POSTGIS_PROJ_VERSION < 60
+
 /**
  * Transform given POINTARRAY
  * from inpj projection to outpj projection
  */
 int
-ptarray_transform(POINTARRAY *pa, projPJ inpj, projPJ outpj)
+ptarray_transform(POINTARRAY *pa, PJ* pj)
 {
 	uint32_t i;
 	POINT4D p;
 
 	for ( i = 0; i < pa->npoints; i++ )
 	{
 		getPoint4d_p(pa, i, &p);
-		if ( ! point4d_transform(&p, inpj, outpj) ) return LW_FAILURE;
+		if ( ! point4d_transform(&p, pj) ) return LW_FAILURE;
 		ptarray_set_point4d(pa, i, &p);
 	}
 
 	return LW_SUCCESS;
 }
 
+int
+lwgeom_transform_from_str(LWGEOM *geom, const char* instr, const char* outstr)
+{
+	char *pj_errstr;
+	int rv;
+	PJ pj;
+
+	pj.pj_from = lwproj_from_string(instr);
+	if (!pj.pj_from)
+	{
+		pj_errstr = pj_strerrno(*pj_get_errno_ref());
+		if (!pj_errstr) pj_errstr = "";
+		lwerror("could not parse proj string '%s'", instr);
```

Comment:
You set `pj_errstr`, which contains the proj error, and then print only the projection string. Could we have both, or at least the error as show by proj?

Reply:
Same in the next conditional.

Reply:
Yeah... no. The issue here is the classic: upstream library has changed their error strings. I found I couldn't pass regression when I was writing out the full error strings. Quick judgement call: strip regression tests or add string?

Reply:
3rd option: rewrite the test so it matches the error with a regex. I think I've done this somewhere in regress/ (but i don't remember where): auxiliar function that receives 2 strings (a query and a regex), executes the query and returns true if an exception is thrown and it matches the error, false otherwise.

Reply:
Please show me where... if it where a return value we could use a postgres regex on it, but since it's an ERROR string, I think we are SOL.

Reply:
It doesn't do any regex, but it might do the trick here: https://github.com/postgis/postgis/blob/515f55bafe0d10d21c199091e3bc500491a72b31/topology/test/regress/st_remedgemodface.sql#L140